Update docs for .netrc integration with provider downloads (#36022)
hashicorp/terraform: website/docs/language/providers/index.mdx

> **Hands-on:** Try the [Lock and Upgrade Provider Versions](/terraform/tutorials/configuration-language/provider-versioning?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.

### Private Providers

> If you have a provider that isn't available in a Hashicorp-hosted registry, such

Reviewer: Making one bigger suggestion for readability. I think this should be two separate sentences to really drive the point home:
```suggestion
If you are using a provider that is not in a Hashicorp-hosted registry, you may need to attach additional credentials to your requests to external registries. You do not need these credentials if your provider is the Terraform public registry or the HCP Terraform private registry.
```

as the Terraform Public Registry or a HCP Terraform Private Registry, you may
need to attach additional credentials to your requests to the external registry.

> By default, Terraform only authenticates the opening request to the registry for

Reviewer: (Suggestion for whole paragraph) I tried rephrasing a few sentences because I thought they flowed a bit better, but if anything isn't technically accurate please feel free to change whatever makes sense to you! 

Also, is there an error folks will receive telling them they need follow-up credentials? If so it might be nice to include that:
```suggestion
By default, Terraform only authenticates the opening request from a provider to the registry. The registry responds with [follow-up URLs](/terraform/internals/provider-registry-protocol#find-a-provider-package) that Terraform makes requests to, such as telling Terraform to download the provider or the `SHASUMS` file. Hashicorp-hosted registries do not require additional authentication for these follow-up requests. If your registry does require additional credentials for follow-up requests, you can use a `.netrc` file to provide those credentials.
```

Author: The error would be whatever is returned by the privately hosted registry which we don't control. It should be a 401 or a 403, but really the owner of the registry can make it do whatever it wants. Hopefully they will have coded the response to be as helpful as possible. There isn't really anything we can say here to make it clear what's happening if that error isn't helpful.

the provider. The registry then responds with
[additional URLs](https://developer.hashicorp.com/terraform/internals/provider-registry-protocol#find-a-provider-package)
that should be followed in order to, for example, download the provider itself
or the `SHASUMS` file. Hashicorp-hosted registries do not require additional
authentication for these requests. If your registry does require additional
credentials for these requests, you can use a `.netrc` file to provide
credentials for these additional requests.

> By default, Terraform searches for the `.netrc` file in your HOME

Reviewer: ```suggestion
By default, Terraform searches for the `.netrc` file in your `HOME` directory. However, you can override the default filesystem location by setting the `NETRC` environment variable. For information on the format of`.netrc`, refer to the [`curl` documentation](https://everything.curl.dev/usingcurl/netrc).
```
